partiql-lang/src/main/kotlin/org/partiql/lang/eval/GlobalsCheck.kt

```diff
@@ -0,0 +1,21 @@
+package org.partiql.lang.eval
+
+interface GlobalsCheck {
+    fun isDefined(name: BindingName): Boolean
+
+    companion object {
+        fun of(session: EvaluationSession): GlobalsCheck = SessionGlobalsCheck(session)
+
+        val empty: GlobalsCheck = EmptyGlobalsCheck()
+    }
+
+    private class SessionGlobalsCheck(val session: EvaluationSession) : GlobalsCheck {
+        override fun isDefined(name: BindingName): Boolean {
+            return session.globals[name] != null
+        }
+    }
```

Comment:
Something about global binding name lookup during compilation time is troubling me, I feel like this should be done in the evaluation time.

Reply:
There is nothing inherently wrong with that.  Such information is commonly provided to compilation by what is commonly known as "static environment", for which `GlobalsCheck` is a stand-in -- until we decide to do that properly, outside of this PR. 

Note that `GlobalsCheck` is an interface and it exposes only static information to the compiler. While an instance of this interface can be created from a run-time object (as it is here), it does not have to be.  

Part of you unease could merely be because the instance-creation code here is in the same file as the interface.  This is certain sloppiness that I might address while working on ideas from John.